docs/conf.py

```diff
@@ -87,3 +90,24 @@ def download_images_from_toml(toml_file, image_dir):
 
 # Download images
 download_images_from_toml(toml_file_path, image_directory)
+
+# -- Fetch external database sources TOML ------------------------------------
+# Download the database_sources.toml hosted on ReadTheDocs
+
+
+def download_database_sources(url, dest_file):
+    content = download_content(url)
+    if content is not None:
+        with open(dest_file, "w") as f:
+            f.write(content)
+
+
+# URL for the hosted external TOML file on ReadTheDocs
+db_sources_url = (
+    "https://readthedocs.org/projects/surfacet-docs/downloads/latest/database_sources.toml"
+)
+# Path to save the database_sources.toml file in docs/
+db_sources_dest = os.path.join(os.path.dirname(__file__), "database_sources.toml")
+
+# Download database sources TOML
+download_database_sources(db_sources_url, db_sources_dest)
```

Comment:
During the ReadTheDocs build, you should be able to just tell it to include the `database_sources.toml` file that exists in the copy of the repository that was cloned as part of the docs build. Fetching from an external URL seems like it would introduce a bit of a chicken and egg problem for making updates to the contents of the file, and make pushing updates to the file more difficult (some manual step if ReadTheDocs even allows that).

Reply:
Either moving it to a new downloads subfolder and setting [html_extra_path](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_extra_path) to include that folder.

Or on one of the docs page add a reference to the toml file in one of the doc pages, with a "download" [role](https://myst-parser.readthedocs.io/en/latest/syntax/roles-and-directives.html#roles-an-in-line-extension-point) could perhaps trick the site build into including the toml file as part of the built site.